src/runtime/virtcontainers/qemu_amd64.go

```diff
@@ -280,11 +319,297 @@ func (q *qemuAmd64) appendProtectionDevice(devices []govmmQemu.Device, firmware,
 				File:            firmware,
 				CBitPos:         cpuid.AMDMemEncrypt.CBitPosition,
 				ReducedPhysBits: cpuid.AMDMemEncrypt.PhysAddrReduction,
+				SevPolicy:       policy,
+				CertFilePath:    godhPath,
+				SessionFilePath: sessionFilePath,
+				KernelHashes:    true,
 			}), "", nil
-	case noneProtection:
-		return devices, firmware, nil
+	} else {
+		return append(devices,
+			govmmQemu.Object{
+				Type:            govmmQemu.SEVGuest,
+				ID:              "sev",
+				Debug:           false,
+				File:            firmware,
+				CBitPos:         cpuid.AMDMemEncrypt.CBitPosition,
+				ReducedPhysBits: cpuid.AMDMemEncrypt.PhysAddrReduction,
+				SevPolicy:       policy,
+			}), "", nil
+	}
+}
 
-	default:
-		return devices, "", fmt.Errorf("Unsupported guest protection technology: %v", q.protection)
+// setup prelaunch attestation
+func (q *qemuArchBase) setupSEVGuestAttestation(ctx context.Context,
+	proxy string,
+	policy uint32) (string, error) {
+
+	logger := virtLog.WithField("subsystem", "SEV attestation")
+	logger.Info("Set up prelaunch attestation")
+
+	certChainBin, err := os.ReadFile(filepath.Join(sevAttestationWorkingDir, sevAttestationCertChainName))
+	if err != nil {
+		log.Fatalf("cert chain not found: %v", err)
+	}
+
+	certChain := b64.StdEncoding.EncodeToString([]byte(certChainBin))
+
+	conn, err := grpc.Dial(proxy, grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		log.Fatalf("did not connect: %v", err)
```

Comment:
Again, if this is an error, why is it not being returned by this function (which can return an error)?

Same comment for code below...